src/sys/socket/mod.rs

```diff
@@ -1202,3 +1206,70 @@ pub fn shutdown(df: RawFd, how: Shutdown) -> Result<()> {
         Errno::result(shutdown(df, how)).map(drop)
     }
 }
+
+#[cfg(any(
+target_os = "linux",
+))]
+#[repr(C)]
+#[allow(missing_debug_implementations)]
+pub struct MMsgHdr<'a>(libc::mmsghdr, PhantomData<&'a ()>);
+
+#[cfg(any(
+target_os = "linux",
+))]
+impl<'a> MMsgHdr<'a> {
+    pub fn new(iov: &mut[IoVec<&'a mut [u8]>], flags: MsgFlags) -> MMsgHdr<'a> {
+        // TODO:
+        //  - support 'control'
+        //  - support 'name'
+        let mut hdr: libc::mmsghdr = unsafe { mem::uninitialized() };
```

Comment:
It looks like you're initializing every one of hdr's fields.  So there's no need to use `mem::uninitialized`.  Just initialize the whole thing at once.

Reply:
I had originally gone that way, but the x86_64-unknown-linux-musl build failed due to missing `_pad1` / `_pad2` initialisation -- https://travis-ci.org/nix-rust/nix/jobs/483633104